sled-agent/src/dump_setup.rs

```diff
@@ -1129,13 +1134,49 @@ impl DumpSetupWorker {
         for path in
             glob::glob(debug_dir.as_ref().join("**/*").as_str())?.flatten()
         {
-            let meta = tokio::fs::metadata(&path).await?;
+            let meta = match tokio::fs::metadata(&path).await {
+                Ok(meta) => meta,
+                // "Not found" errors could be caused by TOCTTOU -- observing a
+                // file which gets renamed, removed, etc. We'll ignore these
+                // files.
+                Err(err)
+                    if matches!(err.kind(), std::io::ErrorKind::NotFound) =>
+                {
+                    continue;
+                }
+                // Other errors get propagated.
+                Err(err) => return Err(err.into()),
+            };
+
+            if !meta.is_file() {
+                // Ignore non-files (e.g. directories, symlinks)
+                continue;
+            }
+
+            // Confirm that the file name is "safe to delete".
+            //
+            // This list may expand as we continue using the Debug dataset.
+            let Some(file_name) = path.file_name().and_then(|n| n.to_str())
+            else {
+                continue;
+            };
+            // Ignore support bundles
+            if file_name == crate::support_bundle::storage::BUNDLE_FILE_NAME {
+                continue;
+            }
+            // Ignore support bundle "temp files" as they're being created.
+            if file_name.ends_with(
+                crate::support_bundle::storage::BUNDLE_TMP_FILE_NAME_SUFFIX,
+            ) {
+                continue;
+            }
```

Comment:
minor: would it be worth turning this into a `fn safe_to_delete(&Path) -> bool` so the concept of support bundles isn't just floating in the middle of `scope_dir_for_cleanup`? (if there're future things that need this sort of immunity, it's likely that'd just get moved into such a thing at that point, so fair enough if we want to punt until then, but it might be that little bit nicer for the next person)